controllers/upgrade.go

```diff
@@ -91,13 +92,22 @@ func (r *RollingUpgradeContext) RotateNodes() error {
 	var (
 		scalingGroup = awsprovider.SelectScalingGroup(r.RollingUpgrade.ScalingGroupName(), r.Cloud.ScalingGroups)
 	)
+	if reflect.DeepEqual(scalingGroup, &autoscaling.Group{}) {
+		r.Info("scaling group not found", "scalingGroup", r.RollingUpgrade.ScalingGroupName(), "name", r.RollingUpgrade.NamespacedName())
+		return errors.Errorf("scaling group not found")
```

Comment:
this will double log, can you just return an error with all the information and remove the info?

Reply:
Yup. That makes sense.